airflow/providers/amazon/aws/operators/athena.py

```diff
@@ -72,19 +82,17 @@ def __init__(  # pylint: disable=too-many-arguments
         self.result_configuration = result_configuration or {}
         self.sleep_time = sleep_time
         self.max_tries = max_tries
-        self.query_execution_id = None
-        self.hook = None
+        self.query_execution_id = None  # type: Optional[str]
 
-    def get_hook(self):
+    @cached_property
```

Comment:
This is great! TIL.

Reply:
:) -- Glad that you found it useful. Tbh I learned it just before I became an Airflow committer -- someone had suggested me in my PR  2-3 years back :)